atoma-service/src/handlers/mod.rs

```diff
@@ -473,54 +472,65 @@ mod vllm_metrics {
     ///
     /// * `client` - The Prometheus HTTP query client.
-    /// * `job` - The Prometheus job name corresponding to the vLLM service instance.
-    /// * `chat_completions_service_url` - The URL of the vLLM service instance, returned alongside the metric.
+    /// * `jobs_with_url` - The Prometheus job names corresponding to the vLLM service instances alongside with the chat completions service url.
     ///
     /// # Returns
     ///
-    /// Returns a `Result` containing a tuple `(String, f64)` on success, where:
+    /// Returns a vector of `Result` containing a tuple `(String, f64)` on success, where:
     ///   - The `String` is the `chat_completions_service_url` passed as input.
     ///   - The `f64` is the calculated 90th percentile of the request queue time in seconds.
     ///
     /// # Errors
     ///
-    /// Returns a `VllmMetricsError` if:
+    /// Return contains a `VllmMetricsError` if:
     ///   - The Prometheus query fails.
     ///   - No metrics data is found for the specified job.
     ///   - The response data cannot be parsed correctly.
-    #[instrument(level = "info", skip_all, fields(job=job))]
+    #[instrument(level = "info", skip_all, fields(jobs_with_url=jobs_with_url.iter().map(|(job, url)| format!("{job}={url}")).collect::<Vec<_>>().join(",")))]
     async fn get_metrics(
         client: &Client,
-        job: &str,
-        chat_completions_service_url: &str,
-    ) -> Result<(String, f64)> {
-        info!(
-            target = "atoma-service",
-            module = "vllm_metrics",
-            level = "info",
-            "Getting metrics for job: {job}"
-        );
+        jobs_with_url: &[(String, String)],
+    ) -> Vec<Result<(String, f64)>> {
         let query = format!(
-            "histogram_quantile(
-                0.90,
-                sum(rate(vllm:request_queue_time_seconds_bucket{{job=\"{job}\"}}[30s])
-                    or vector(0)) by (le)
-            )"
+            "histogram_quantile(0.90, sum by (le,job) (rate(vllm:request_queue_time_seconds_bucket{{job=~\"{jobs}\"}}[30s])))",
+            jobs = jobs_with_url
+                .iter()
+                .map(|(job, _url)| job.as_str())
+                .collect::<Vec<_>>()
+                .join("|")
         );
-        let response = client.query(&query).get().await?;
-        response.data().as_vector().map_or_else(
-            || Err(VllmMetricsError::NoMetricsFound(job.to_string())),
-            |data_vector| {
-                data_vector.first().map_or_else(
-                    || Err(VllmMetricsError::NoMetricsFound(job.to_string())),
-                    |value| {
-                        let sample = value.sample();
-                        let value = sample.value();
-                        Ok((chat_completions_service_url.to_string(), value))
-                    },
-                )
-            },
-        )
+        let response = client.query(&query).get().await;
+        jobs_with_url
```

Comment:
Do we need to change the way the config.toml is structured ?

Reply:
no, config is good